modules/Rock.cmake

```diff
@@ -1205,6 +1215,35 @@ macro (rock_find_qt4)
     endforeach()
 endmacro()
 
+# Find Qt5 libraries
+#
+# The macro finds QtCore, QtGui and QtOpenGL by default. Additional arguments
+# can be given to find more components.
+macro (rock_find_qt5)
+    set(__arglist "${ARGN}")
+    list(GET 0 __arglist __arg_optreq)
+    if ((__arg_optreq EQUAL "OPTIONAL") OR (__arg_optreq EQUAL "REQUIRED"))
+        list(REMOVE_AT __arglist 0)
+    else()
+        set(__arg_optreq REQUIRED)
+    endif()
+
+    find_package(Qt5 ${__arg_optreq} COMPONENTS Core Gui OpenGL ${__arglist})
+    include_directories(${QT_HEADERS_DIR})
+    set(ROCK_QT_VERSION 5)
+    
+    foreach(__qtmodule__ Core Gui OpenGL ${ARGN})
+        #string(TOUPPER ${__qtmodule__} __qtmodule__)
+        add_definitions(${Qt5${__qtmodule__}_DEFINITIONS})
+        include_directories(${Qt5${__qtmodule__}_INCLUDE_DIRS})
+        link_directories(${Qt5${__qtmodule__}_LIBRARY_DIR})
+    endforeach()
```

Comment:
The behavior of forcing the addition of some modules in rock_find_qt4 is ... questionable (at best). Maybe we should not replicate it in the new rock_find_qt5 ? What do you think ?

Reply:
Additionally, I would guess that it would be best here to use the CMake imported targets instead of the old-style add_definitions/include_directories/link_directories. `rock_find_qt4` existed from before that time ;-)

Reply:
Yes, this is a good point. I think we shouldn't enforce any kind of library dependency here. It is also possible to only use e.g. Qt5::Widgets, so I'm going to remove the hardcoded deps.

Reply:
I completely removed the old-fashioned include and link directories. They are not necessary as dependencies are set up correctly via the DEPS argument in rock_executable and rock_library.